Remove Universal Analytics

martynmjones opened the pull request:
### Changes proposed in this Pull Request:

Closes #215

- [x] #309
- [x] #320
- [x] #311
- [x] #321
- [x] #328

### Checks:
* [x] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changelog entry

> Remove - Universal Analytics integration

Commit:
Add settings field

Commit:
Add option defaults

Commit:
Restructure get_product_identifier

Commit:
Add unit tests

martynmjones committed:
Add default option value

martynmjones committed:
Set default options for new installations

martynmjones committed:
Fix CS

Commit:
Fix CS

Commit:
Update product object structure

Commit:
Update view_item_list event

Commit:
Update add_to_cart event

Commit:
Update remove_from_cart event

Commit:
Add begin_checkout event

Commit:
Remove email, phone, and billing address events

Commit:
Add add_shipping_info event

Commit:
Remove trackCheckoutStep

Commit:
Track add_shipping_info on checkout submit

Commit:
Remove trackCheckoutOption

Commit:
Remove add_payment_info event

Commit:
Update select_content event

Commit:
Remove @wordpress/i18n import

Commit:
Remove currentStep

Commit:
Export getProductId

Commit:
Update comment

martynmjones committed:
Merge pull request #309 from woocommerce/add/295-product-identifier-setting

Add product identifier setting

Commit:
Check data exists before tracking view_item_list

Commit:
Check product for quantity

Commit:
Remove tracking from class-wc-google-analytics.php

Commit:
Remove UA settings

Commit:
Remove class-wc-google-analytics-js.php

Commit:
Remove admin-ga-settings.js

Commit:
Remove unused alias

Commit:
Remove UA tracking from abstract class

Commit:
Move tracking to class-wc-google-gtag-js.php

Commit:
Fix unit tests

Commit:
Add notice if property ID starts with "UA"

Commit:
CS Fixes

Commit:
Fix event name

Commit:
Re-order settings fields

Commit:
Remove event_tracking_code

Commit:
Fix alignment

martynmjones committed:
Merge pull request #320 from woocommerce/remove/ua-code

Remove Universal Analytics code

martynmjones committed:
Escape notice text and update external URL

martynmjones committed:
Merge pull request #321 from woocommerce/add/ua-update-notice

Add upgrade notice if property ID starts with "UA"

Commit:
Update formatPrice to return a number

Commit:
Conditionally include coupon

martynmjones committed:
Add cart coupon utility

martynmjones committed:
Merge pull request #311 from woocommerce/update/268-tracking-code-for-blocks

Update tracking for blocks

Commit:
Remove inline JS tracking

Commit:
Remove abstract methods

Commit:
Add inline script data

Commit:
Format and attach event data in abstract class

Commit:
Map classic hooks to Blocks and inline event data

Commit:
Update wcgaiData single product name

Commit:
Remove key for wcgaiData products

Commit:
Fix array keys

Commit:
Update event map and script data

Commit:
Add a tracker utility

Commit:
Update utils

Commit:
Move actions to data formatting functions

Commit:
Add tracker utility

Commit:
Add blocks and classic integrations

Commit:
Add config file

Commit:
Add new entry point

Commit:
Remove enqueue_standard_tracking_code

Commit:
Update method signature

Commit:
Remove event map

Commit:
Rename data formatters

Commit:
Restructure Tracker

Commit:
Update tracking

Commit:
Reduce categories data to name only

Commit:
Update event => WC hook map

Commit:
Update Tracker

Commit:
Add utility to find a single product by its ID

Commit:
Allow for single entry for data type in script data

Commit:
Track add to cart events

martynmjones committed:
Fix JS formatting issues

martynmjones committed:
Remove unused PHP tests

martynmjones committed:
Update composer lock file

martynmjones committed:
Update composer.json

martynmjones committed:
Remove calls to deleted functions

martynmjones committed:
Update cart formatting

martynmjones committed:
Add remove_from_cart tracking

martynmjones committed:
Add begin_checkout tracking

martynmjones committed:
Add select_content tracking

martynmjones committed:
Add purchase tracking

martynmjones committed:
Fix param description

martynmjones committed:
Fix price formatting

martynmjones committed:
Add check for enabled events

martynmjones committed:
Use correct product identifier

martynmjones committed:
Add product identifier to cart item

martynmjones committed:
Add unit tests

martynmjones committed:
Refactor test_script_data to support previous versions of PHP

martynmjones committed:
Refactor getProductId util

martynmjones committed:
Add support for product variations

martynmjones committed:
Update plugin name to Google Analytics for WooCommerce

martynmjones committed:
Additional name updates

martynmjones committed:
Merge pull request #353 from woocommerce/update/branding-to-google-analytics-for-woocommerce

Update plugin branding to Google Analytics for WooCommerce

tomalec committed:
Simplify tracker event handler API,

Remove not used "event" object.

Addresses https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/328#discussion_r1483531223

martynmjones committed:
Merge pull request #357 from woocommerce/update/tracking-for-classic-pages-simplified-tracker

Simplify tracker event handler API

martynmjones committed:
Fix price formatting

martynmjones committed:
Remove unnecessary check

martynmjones committed:
Remove demo comment

martynmjones committed:
Assign formatter to constant inside eventHandler

martynmjones committed:
Simplify set_script_data

martynmjones committed:
Rename tracker_var to tracker_function_name

martynmjones committed:
Update unit tests

martynmjones committed:
Update config export

martynmjones committed:
Fix addedToCart declaration

martynmjones committed:
Remove spread operator

martynmjones committed:
Update classic select_content tracking

martynmjones committed:
Update remove_from_cart tracking and add mini-cart support

martynmjones committed:
Update select_content selector

martynmjones committed:
Add comment describing why some actions are removed

martynmjones committed:
Fix linting errors

tomalec committed:
Remove options remapping, use settings directly.

Fix https://github.com/woocommerce/woocommerce-google-analytics-integration/issues/359

tomalec committed:
Remove redundant WC_Google_Analytics properties

use `$this->settings` directly.

martynmjones committed:
Fix ID mismatch when removing a variation from cart

martynmjones committed:
Adjust append_script_data

tomalec committed:
Rename options to settings where applicable.

tomalec committed:
Merge branch 'update/tracking-for-classic-pages' into dev/refactor-options

martynmjones committed:
Avoid overwriting third-party event handlers

Co-authored-by: Tomek Wytrębowicz <tomalecpub@gmail.com>

martynmjones committed:
Adjust select_content conditions and update comment

martynmjones committed:
Fix spacing

martynmjones committed:
Merge pull request #328 from woocommerce/update/tracking-for-classic-pages

Update tracking for classic pages

tomalec committed:
Implement MVP for consent mode support.

Deny all for EEA visitors. Document the customization snippet.

tomalec committed:
Update custom consent snippet

to cover the quirks of wcgaiData not being loaded otherwise.

tomalec commented:
I found an issue that when you add (in a snippet or another extension) another script that has a dependency on `woocommerce-google-analytics-integration` and enqueue it not in the footer, for some reason the inline script with `wcgaiData` is not rendered, breaking the functionality.

### Steps to reproduce
1. Add a snippet:
```php
add_action( 'wp_enqueue_scripts', function () {
    wp_register_script( 'my-code', '', array('woocommerce-google-analytics-integration'), null, false );
    wp_add_inline_script( 'my-code', "console.log('Hello');" );
    wp_enqueue_script( 'my-code' );
} );
```
2. open shop page

tomalec committed:
Set `wp_enqueue_scripts` priority

Co-authored-by: martynmjones <martynmjones@gmail.com>

tomalec committed:
Dequeue the WC Blocks GA integration srcipt

not to let it register its `gtag` function so that we could provide a more detailed configuration.

Co-authored-by: martynmjones <martynmjones@gmail.com>

tomalec committed:
Process tracker config separately from feeding data to classic tracking

Set up tracker synchronously once the script is loaded (possibly in the `<head>`).
Track classic WC events only after data is ready and the document is loaded - at leas on `wp_footer`.

Address https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/322#issuecomment-1967508597

tomalec committed:
Merge pull request #363 from woocommerce/fix/split-config

tomalec committed:
Merge branch 'remove-universal-analytics' into add/consent-mode-v0

tomalec committed:
Update README

after https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/363

tomalec committed:
Fix the consent region code from `UK` to `GB`

Co-authored-by: martynmjones <40762232+martynmjones@users.noreply.github.com>

tomalec committed:
Merge pull request #360 from woocommerce/dev/refactor-options

tomalec committed:
Merge pull request #361 from woocommerce/add/consent-mode-v0

martynmjones committed:
Resolve merge conflicts

martynmjones committed:
PHPCS

martynmjones committed:
Fix JSDoc spacing
